blaze-client/src/main/scala/org/http4s/client/blaze/bits.scala

```diff
@@ -21,12 +21,13 @@ private[blaze] object bits {
 
   val ClientTickWheel = new TickWheelExecutor()
 
-  def getExecutor(config: BlazeClientConfig): (ExecutorService, Task[Unit]) = config.customExecutor match {
-    case Some(exec) => (exec, Task.now(()))
-    case None =>
-      val exec = threads.newDaemonPool("http4s-blaze-client")
-      (exec, Task.delay(exec.shutdown()))
-  }
+  def getExecutionContext(config: BlazeClientConfig): (ExecutionContext, Task[Unit]) =
+    config.customExecutionContext match {
+      case Some(ec) => (ec, Task.now(()))
+      case None =>
+        val exec = threads.newDaemonPool("http4s-blaze-client")
```

Comment:
So, if I understand correctly. It sounds like standardizing on `ExecutionContext.global` where it is used and getting rid of the option at the edges. 

Then like everything on this branch, get functional first and then we will see where performance improvements can be made.